<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MPSDPulser</class>
 <widget class="QDialog" name="MPSDPulser">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>453</width>
    <height>181</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Pulser ... </string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="pulserGroupBox">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="manualPulser">
      <attribute name="title">
       <string>Manual</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0" colspan="3">
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="textLabel1_4_2">
           <property name="text">
            <string>MCPD ID</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="MCPDSpinBox" name="devid">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="maximum">
            <number>0</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="textLabel1_4_2_2">
           <property name="text">
            <string>Module
(Bus-#)</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="ModuleSpinBox" name="module">
           <property name="maximum">
            <number>7</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="textLabel1_4_2_3_4">
           <property name="text">
            <string>Channel</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="pulsChan">
           <property name="toolTip">
            <string>pulser channel</string>
           </property>
           <property name="maximum">
            <number>7</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QCheckBox" name="pulserButton">
             <property name="toolTip">
              <string>toggle pulser on/off</string>
             </property>
             <property name="text">
              <string>On/Off</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item row="1" column="1">
        <widget class="QGroupBox" name="buttonGroup5">
         <property name="title">
          <string>Amplitude (0...255)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_17">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QRadioButton" name="pulsampRadio1">
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="pulsAmp1">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="pulsampRadio2">
              <property name="text">
               <string/>
              </property>
              <property name="checked">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="pulsAmp2">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>90</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QGroupBox" name="positionGroup">
         <property name="title">
          <string>Position</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_11">
          <item row="0" column="0">
           <widget class="QRadioButton" name="pulsLeft">
            <property name="text">
             <string>left</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QRadioButton" name="pulsMid">
            <property name="text">
             <string>mid</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QRadioButton" name="pulsRight">
            <property name="text">
             <string>&amp;right</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="automaticPulser">
      <attribute name="title">
       <string>Automatic Test</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="1" column="2">
        <widget class="QPushButton" name="automaticPulserTest">
         <property name="text">
          <string>Automatic pulser test</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QCheckBox" name="infiniteBox">
         <property name="text">
          <string>infinite</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLed" name="qLed" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>50</width>
           <height>50</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Binary Led</string>
         </property>
         <property name="whatsThis">
          <string>Led widget</string>
         </property>
         <property name="value" stdset="0">
          <bool>false</bool>
         </property>
         <property name="blink" stdset="0">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QGroupBox" name="buttonGroup5_2">
         <property name="title">
          <string>Test amplitude (0...255)</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_18">
          <item row="0" column="0">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QSpinBox" name="testAmp1">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>30</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="testAmp2">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>60</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QLed</class>
   <extends>QWidget</extends>
   <header>qled.h</header>
  </customwidget>
  <customwidget>
   <class>ModuleSpinBox</class>
   <extends>QSpinBox</extends>
   <header>ModuleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>MCPDSpinBox</class>
   <extends>QSpinBox</extends>
   <header>MCPDSpinBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>module</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>displayMPSDSlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>305</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>devid</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setMCPD(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulserButton</sender>
   <signal>stateChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setPulser(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>22</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>101</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsAmp1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>amplitudeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsAmp2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>amplitudeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>263</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>547</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsLeft</sender>
   <signal>clicked(bool)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setPulserPosition(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>550</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsMid</sender>
   <signal>clicked(bool)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setPulserPosition(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>387</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>547</x>
     <y>64</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsRight</sender>
   <signal>clicked(bool)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setPulserPosition(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>445</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>546</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pulsChan</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MPSDPulser</receiver>
   <slot>setChannel(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>458</x>
     <y>29</y>
    </hint>
    <hint type="destinationlabel">
     <x>274</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>devid</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>412</x>
     <y>170</y>
    </hint>
    <hint type="destinationlabel">
     <x>153</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>module</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>305</x>
     <y>28</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsChan</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>458</x>
     <y>29</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsampRadio1</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>buttonGroup5</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>248</x>
     <y>126</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsLeft</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>333</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsMid</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>387</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulsRight</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>445</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>pulserButton</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>30</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>automaticPulserTest</sender>
   <signal>toggled(bool)</signal>
   <receiver>qLed</receiver>
   <slot>setValue(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>444</x>
     <y>183</y>
    </hint>
    <hint type="destinationlabel">
     <x>53</x>
     <y>172</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MPSDPulser</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>235</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>-11</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>displaySlot(int)</slot>
  <slot>setChannel(int)</slot>
  <slot>setMCPD(int)</slot>
  <slot>amplitudeChanged(int)</slot>
  <slot>setPulserPosition(bool)</slot>
  <slot>setPulser(bool)</slot>
 </slots>
</ui>
